sys/include/psa_crypto/psa/error.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2024 TU Dresden
+ * Copyright (C) 2021 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_psa_crypto
+ * @{
+ *
+ * @file        error.h
+ * @brief       Error definitions for the PSA Crypto API
+ *
+ * @details     This header file is also compatible with the PSA Certified Status code API.
+ *
+ * @author      Armin Wolf <wolf.armin@mailbox.tu-dresden.de>
+ * @author      Lena Boeckmann <lena.boeckmann@haw-hamburg.de>
+ *
+ */
+
+#ifndef PSA_CRYPTO_PSA_ERROR_H
+#define PSA_CRYPTO_PSA_ERROR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+/**
+ * @brief   Status code type used for all PSA Certified APIs.
+ *
+ * @details This is either @ref PSA_SUCCESS, which is zero, indicating success; or a small
+ *          negative value indicating that an error occurred. Errors are encoded as one of
+ *          the @c PSA_ERROR_xxx values defined here.
+ */
+#ifndef PSA_SUCCESS
```

Comment:
Nice, thank you!
One question: Why did you add this guard? Is that necessary?

Reply:
The PSA status code API says:

>Implementation note
>
>This definition is permitted to be present in multiple header files that are included in a single compilation.
>
>To prevent a compilation error from duplicate definitions of [psa_status_t](https://arm-software.github.io/psa-api/>status-code/1.0/api/status-codes.html#c.psa_status_t), the definition of [psa_status_t](https://arm-software.github.io/>psa-api/status-code/1.0/api/status-codes.html#c.psa_status_t) must be guarded, by testing for an exiting definition of >[PSA_SUCCESS](https://arm-software.github.io/psa-api/status-code/1.0/api/status-codes.html#c.PSA_SUCCESS), in any >header file that defines [psa_status_t](https://arm-software.github.io/psa-api/status-code/1.0/api/status->codes.html#c.psa_status_t).
>
>The definition of [psa_status_t](https://arm-software.github.io/psa-api/status-code/1.0/api/status->codes.html#c.psa_status_t) above shows the recommended form of the guard.

Reply:
Ah nice, I think that's new, then.